.github/workflows/pr-e2etest.yml

```diff
@@ -0,0 +1,46 @@
+name: PR e2e test
+
+on:
+  workflow_call:
+
+jobs:
+  pr_e2e_tests:
+    runs-on: namespace-profile-default
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 'stable'
+
+      - name: Build halo and relayer binaries
+        uses: goreleaser/goreleaser-action@v5
+        env:
+          GOOS: linux
+        with:
+          version: latest
+          args: build --single-target --snapshot --clean --id=halo --id=relayer
+
+      - name: Build halo image
+        run: |
+          cd dist/halo_linux_amd64_v1
+          docker build -f "../../halo/Dockerfile" . -t "omniops/halo:${GITHUB_SHA::7}"
+
+      - name: Build relayer image
+        run: |
+          cd dist/relayer_linux_amd64_v1
+          docker build -f "../../relayer/Dockerfile" . -t "omniops/relayer:${GITHUB_SHA::7}"
+
+      - name: Run devnet1 e2e test
+        run: |
+          go install github.com/omni-network/omni/test/e2e
+          cd test/e2e && ./run-multiple.sh manifests/devnet1.toml
+
+      - name: Upload failed logs
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: failed-logs
+          path: test/e2e/failed-logs.txt
+          retention-days: 3
```

Comment:
Should we upload the failed logs directly to Loki instead?
There is a [loki-action](https://github.com/marketplace/actions/loki-action) available.

Reply:
```yaml
    - name: loki-logs
      if: failure()
      uses: metrico/loki-action@V3.1
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        endpoint: ${{ secrets.LOGQL_ENDPOINT }}
        username: ${{ secrets.LOGQL_USER }}
        password: ${{ secrets.LOGQL_PASS }}
        job-names: build
```

Reply:
Good idea!